### DISCUSSION TITLE
Is there any .d.ts file for Zepos api?

### DISCUSSION THREAD

**Discussion by meteozond** [zepp-health/zeppos-forum#164] [Q&A]
I think it could be useful during app and watchface development.

**Comment by htoooth** [comment]
@meteozond  /gpt 
通过工具创建应用，项目内部就有声明文件。 声明文件没有独立仓库。

When creating applications through tools, the declaration files are included within the project. There is no separate repository for declaration files.

**Reply by meteozond** [reply]
@htoooth Which tools do you mean? [Zeus CLI](https://docs.zepp.com/docs/guides/tools/cli/)? If no - there is no definitions...

**Reply by htoooth** [reply]
@meteozond 
是的。你通过 zeus 创建项目之后，有一个 global.d.ts 文件，这里面就是类型声明。
Yes. After you create a project through Zeus, there is a global.d.ts file that contains type declarations.

**Reply by meteozond** [reply]
Got it! But as I see most of primitives defined under some intermediate namespaces, when it usually accessed directly:
<img width="359" alt="image" src="https://github.com/zepp-health/zeppos-forum/assets/975689/2af84103-05fc-42bf-a906-916e9cc86ff6">

**Reply by meteozond** [reply]
Sorry, I've missed at the end of namespace
```js
  let hmUI: HmWearableProgram.DeviceSide.HmUI.IHmUI
  export = hmUI
```
Otherwise, Intellij idea doesn't see it for some reason :(

**Reply by meteozond** [reply]
I've found that most of constants are missing 
<img width="507" alt="image" src="https://github.com/zepp-health/zeppos-forum/assets/975689/a5f2950c-4866-4895-9109-c6d1941b6e92">

**Reply by MyoungXUE** [reply]
@meteozond 
Hi, the statement document does have omissions.

The declaration file for 1.0 will not be officially maintained at this time.

We will update the declaration file for 2.x and later. The declaration file is for reference only, more information is subject to the developer documentation.

**Reply by meteozond** [reply]
@MyoungXUE Documentation is definitely useful, but is much more handy when you can access it from the code :(
